### DISCUSSION TITLE
Dynamic k8s pod resources

### DISCUSSION THREAD

**Discussion by Tehada** [dagster-io/dagster#8001] [Q&A] [answered]
According to [docs](https://docs.dagster.io/deployment/guides/kubernetes/customizing-your-deployment), we can specify resources for op's pod using syntax below:

```python
@op(
    tags={
        "dagster-k8s/config": {
            "container_config": {
                "resources": {
                    "requests": {"cpu": "200m", "memory": "32Mi"},
                }
            },
        }
    }
)
def my_op(context):
    context.log.info("running")

@job(executor_def=k8s_job_executor)
def my_job():
    my_op()
```

But what if I want to set resources for pod at runtime, as a result of upstream's op calculation? Is it possible? My usecase is to have a task, which lists gcs bucket for new files (amount of files and their sizes are variable day-to-day) and then launches pod to process each file with custom resources values -- some files require 20 GiB of RAM, while others can take up to 100 GiB.

**Comment by smackesey** [comment]
cc @gibsondan @rexledesma @johannkm

**Comment by johannkm** [comment, marked as answer]
This isn't currently supported- we don't have a great place in the API to specify it. Created tracking issue https://github.com/dagster-io/dagster/issues/8062

**Comment by slamer59** [comment]
Hello, I am also interested. 
I wonder if you could split the graphs in two. 

The first graph make the calculation and the last op is Trigering a job based on this resource. 

Let's say with https://docs.dagster.io/_apidocs/schedules-sensors#dagster.RunRequest ?

```python
yield RunRequest(
    run_key=filename,
    run_config={...},
    tags={
        "dagster-k8s/config": {
            "container_config": {
                "resources": {
                    "requests": {"cpu": "**200m**", "memory": "**32Mi**"},
                }
            },
        }
    }
)
```

I would try this as a work around.
